### ISSUE TITLE
How can I have custom identifiers/primary keys for my resources?

### ISSUE THREAD

**Issue by akamit21** [Steams/ra-data-hasura-graphql#57]
My table has a primary key other than id and react-admin enforces id to be returned in the response by the DataProvider. So can I configure different primary keys/identifiers for my resources?

**Comment by affluent-bilby-classifieds** [comment]
hi Akamit21,

I am a novice (at least as a developer) myself but I hope this helps you. 

In the readme for ra-data-hasura it has a section on [different primary keys](https://github.com/hasura/ra-data-hasura#different-primary-keys)

best regards,

Chris

**Comment by avimoondra** [comment]
Have a similar question as @akamit21 !

(@affluent-bilby-classifieds that is for a different library that is not compatible with ra-data-hasura-graphql. I'm actually in the process of migrating from that library to this one.)

**Comment by avimoondra** [comment]
A guess would be some modification here: https://github.com/Steams/ra-data-hasura-graphql#customize-fields-variables-responseparser ?

**Comment by akamit21** [comment]
> Have a similar question as @akamit21 !
> 
> (@affluent-bilby-classifieds that is for a different library that is not compatible with ra-data-hasura-graphql. I'm actually in the process of migrating from that library to this one.)

Yes, I tried to implement but didn't work out.

**Comment by polymeris** [comment]
Having the same issue, with the added complication that I am using composite primary keys. Did any of you figure it out?

**Comment by akamit21** [comment]
Nope, not yet, but to resolve my primary key problem I rename the `primary key` column to `id` when it fetches the data and generates the data-list, in that way I receive my `primary-key` column renamed as `id`

**Comment by divramod** [comment]
@akamit21 can you show an example on how you are renaming the primary key to id while fetching the data?
it would be nice to have a option to configure that.

**Comment by diazhector98** [comment]
@divramod agreed. @akamit21 could you show an example?

**Comment by akamit21** [comment]
Well, I made the changes in `getResponseParser.js` file of the library. I had a employee table with `primary_key` name `emoloyee_id`. So I add only these line in code and it worked.

`if (key === 'employee_id') {
      return { ...acc, id: dataKey };
 }`
